cli/src/args.rs

```diff
@@ -38,10 +38,15 @@ pub struct Args {
     /// context, if any.
     pub token: Option<String>,
 
+    #[structopt(long = "proxy")]
+    /// URL for an HTTP proxy that will be used for all requests if specified
+    pub proxy: Option<Option<Url>>,
```

Comment:
a context optionally has a proxy and the argument is itself option, so `Option<Option<Url>>` -- e.g.
```
re get comments // uses proxy from context; proxy == None
re --proxy get comments // overwrites the proxy from the context to use no proxy; proxy == Some(None)
re --proxy http//example.proxy // overwrites the proxy from the context to use http//example.proxy; proxy == Some(Some("http//example.proxy"))
```